### PR TITLE
[ML] Refactoring inference HTTP Client to allow dynamic settings updates

### PR DESCRIPTION
**Pull request by jonathan-buttner**
This PR is a follow on for https://github.com/elastic/elasticsearch/pull/100359. This PR allows the connection pull and eviction settings to be updated.

This PR also queues requests to the http client since submitting a request can block.

### CODE REVIEW

File: x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/external/http/HttpClient.java [hunk 1]
```diff
@@ -88,24 +60,32 @@ private static CloseableHttpAsyncClient createAsyncClient(PoolingNHttpClientConn
...
-            connectionEvictor.start();
         }
     }
 
-    public void send(HttpUriRequest request, ActionListener<HttpResult> listener) throws IOException {
+    public void send(HttpUriRequest request, ActionListener<HttpResult> listener) {
         // The caller must call start() first before attempting to send a request
         assert status.get() == Status.STARTED;
 
+        threadPool.executor(HTTP_CLIENT_SENDER_THREAD_POOL_NAME).execute(() -> {
```
Comment (jonathan-buttner) [inline x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/external/http/HttpClient.java:79]: Would it be better to build our own executor service instead? One advantage I can see for building our own is defining a limit to the queue. I couldn't find a way to do that in the `ScalingExecutorBuilder`. Another option would be to use a `FixedExecutor` but that'll force at least 1 thread to be waiting around for tasks I think.

File: x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/external/http/HttpSettings.java [hunk 2]
```diff
@@ -26,89 +25,24 @@ public class HttpSettings {
         Setting.Property.NodeScope,
         Setting.Property.Dynamic
     );
-    static final Setting<Integer> MAX_CONNECTIONS = Setting.intSetting(
```
Comment (jonathan-buttner) [inline x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/external/http/HttpSettings.java:29]: These were moved up into the `HttpClientManager`.

File: x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/external/http/HttpClient.java (hunk 1 above)
Comment (davidkyle) [reply x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/external/http/HttpClient.java:79]: Callers of this method could implement a queue if necessary and the callee is better placed to make decisions about retries etc. The important thing is that `doPrivilegedSend` can block acquiring a connection so the work should be done by the executor.

**Review by davidkyle — APPROVED** [review]
LGTM

File: x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/external/http/HttpClient.java (hunk 1 above)
Comment (jonathan-buttner) [reply x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/external/http/HttpClient.java:79]: Good point, I'll change this in a follow up PR 👍

**Pull request merged**